internal/http.go

```diff
@@ -63,6 +65,22 @@ func httpDo(client *http.Client, method, url string, headers map[string]string,
 	return readResponse(res)
 }
 
+func httpDo(client *http.Client, method, url string, headers map[string]string, data interface{}) (*httpRes, error) {
+	var err error
+	var res *httpRes
+	for _, delay := range []int{0, 1, 1, 2, 4, 8, 16} {
```

Comment:
Another nitpick:

I usually prefer a simplified semi-fibonacci for things like this i.e. `0, 1, 2, 5, 13, 30`.

Rationale: In case of server errors those 4 retry attempts (`0, 1, 1, 2`) at the start of your loop will usually only make things worse by creating additional load onto a struggling server; and have nearly zero chance to succeed if there is a problem.

Starting with bigger retry intervals and quickly going up a higher final value (arguably) gives the retry logic a better chance to succeed, while creating less load onto the server.

Reply:
it might have been a dumb approach - and at a minimum i didn't explain. i kind of went between Fibonacci and exponential. But if you are inventing something - it's probably dumb or someone else would have done it.

Reply:
updated in force-push